prediction_market_agent/agents/crewai_subsequential_agent/deploy.py

```diff
@@ -0,0 +1,71 @@
+import random
+from decimal import Decimal
+
+from prediction_market_agent_tooling.deploy.agent import DeployableAgent
+from prediction_market_agent_tooling.markets.agent_market import AgentMarket
+from prediction_market_agent_tooling.markets.data_models import BetAmount, Currency
+from prediction_market_agent_tooling.markets.markets import MarketType
+
+from prediction_market_agent.agents.crewai_subsequential_agent.crewai_agent_subquestions import (
+    CrewAIAgentSubquestions,
+)
+from prediction_market_agent.agents.utils import market_is_saturated
+
+
+class DeployableThinkThoroughlyAgent(DeployableAgent):
+    # For cheaper credits at this experimental stage
+    def __init__(self) -> None:
+        super().__init__()
+
+    def pick_markets(self, markets: list[AgentMarket]) -> list[AgentMarket]:
+        # We simply pick 5 random markets to bet on
+        picked_markets: list[AgentMarket] = []
+        random.shuffle(markets)
+        for market in markets:
+            # Assume very high probability markets are already known, and have
+            # been correctly bet on, and therefore the value of betting on them
+            # is low.
+            if not market_is_saturated(market=market):
+                picked_markets.append(market)
+                if len(picked_markets) == 5:
+                    break
+
+        return picked_markets
+
+    def answer_binary_market(self, market: AgentMarket) -> bool:
+        # The answer has already been determined in `pick_markets` so we just
+        # return it here.
+        result = CrewAIAgentSubquestions().answer_binary_market(market.question)
+        return True if result.decision == "y" else False
```

Comment:
So if all the tasks above outputs `p_yes=0.5, p_no=0.5, confidence=0, decision=""`, then this will default to False?

Is that the desired behaviour?

Reply:
So now, per your suggestion, the return statement looks like this
```
return True if result.decision == "y" else False if result.decision == "n" else should_not_happen()
```
so if it errors out and no decision can be made, then we throw an error, because the agent couldn't reason about a decision.

Do you think this should be changed @kongzii ?

Reply:
> so if it errors out and no decision can be made, then we throw an error, because the agent couldn't reason about a decision.

Yeah, if the agent fails we can debug. But if he silently bets on NO the entire time, we won't know about it.

Reply:
Cool, let's keep as it's implemented now and debug as needed

Reply:
What's the plan for debugging if we don't know it's failing because it will default to False?

Reply:
Another option, and I actually like this one more, is to do:

```
class ProbabilityOutput(BaseModel):
    decision: t.Literal["y","n"]
    p_yes: float
    p_no: float
    confidence: float
```

(use the Literal typing instead of just string)

Reply:
Edit: Nevermind! I'm just commenting too fast, sorry 🤦